### ISSUE TITLE
Restore download snackbar/logic for 2 zoomlevel-"download full size" offer

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/android#2598]
Likely due to #2593 we lost this feature.
I just checked with @tobiasKaminsky and the dev build from two days ago and it in fact does show a snackbar for downloading the full-size version of the image. This feature is now lost due to the referenced PR. My guess is that this happened due to the "new" zoom logic and would need to get re-wired again.

cc @ardevd: Any ideas how to fix this?

**Comment by ardevd** [comment]
@AndyScherzinger could you provide me with a step by step instruction on how to reproduce the issue and a description on what's supposed to happen?

**Comment by AndyScherzinger** [comment]
@ardevd: Sure thing :)

1. open an image which hasn't been downloaded --> shows a preview image (lower resolution) from server
2. zoom-in two times --> a snackbar should show up asking if you want to download the full-resolution images

In the past I think the zoom-to-download part has been implemented via #1710

**Comment by ardevd** [comment]
@AndyScherzinger hmm, I can't really replicate this behaviour with the app before my PR. Firstly there is only one zoom level. Secondly, I don't get a snackbar message at all when zooming in. 

The exception seems to be png files with transparent backgrounds where I immediately get a snackbar message saying "No resized image available. Download full image?"

So I can't really see any difference here before and after my PR, but I might be missing something.

**Comment by AndyScherzinger** [comment]
Maybe @tobiasKaminsky can clarify how to reproduce the behavior tomorrow.

**Comment by AndyScherzinger** [comment]
@ardevd My mistake... the snackbar for full-size image download still shows up... I could reproduce it even with your introduced photoview via: pinch-to-zoom-in "like crazy"... then the snackbar comes up :)

**Comment by ardevd** [comment]
So should we close this issue or rephrase it? :)

**Comment by AndyScherzinger** [comment]
I rephared it :)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @ardevd for whatever reason, this lint issue comes up since today:
```
<string name="resized_images_download_full_image">Download full image?</string>
```
is now an unused resource, I then checked the pinch-to-zoom-to-download-snackbar again and it doesn't show up anymore! :warning:

**Comment by ardevd** [comment]
@AndyScherzinger That's odd. I'm pretty sure I didnt touch it and it worked fine after #2593 so there must have been a merge after that. Time to check the logs ;)

**Comment by ardevd** [comment]
@AndyScherzinger `R.string.resized_images_download_full_image` was used in `TouchImageViewCustom.java`, so that explains why its not used anymore. However, a similar message did show up after #2593 was merged. I'll look into it.

**Comment by ardevd** [comment]
I can confirm the same. I'm developing a new implementation for it now. Double tapping the image will bring up the snackbar message prompting the user to download the image.

**Comment by ardevd** [comment]
However, perhaps we can do this in a more elegant user? As a NextCloud user for years, I never noticed the prompt until working on #2593. Perhaps we can expose the feature in a smarter way?

**Comment by AndyScherzinger** [comment]
The feature isn't easy to discover while I personally don't have a good idea on how we could improve its discoverability or if like you said we don't need it at all since people can use the menu to download while the snackbar logs the Yeses and then auto-downloads so it has some more logic to it than just prompt+download.

What do you think @tobiasKaminsky (original author), @mario and @nextcloud/designers ?

**Comment by ardevd** [comment]
I'm generally in favor of scrapping the feature as it doesnt really make sense from a UX perspective in my opinion and acts more like a random gesture control that most user will at best accidentally stumble upon. Also, since it's not a UX pattern used anywhere else, it's not exactly a defining feature.

But interested to hear what you guys think

**Comment by tobiasKaminsky** [comment]
As far as I remember it was an idea coming up while limiting resized images to zoom factor 2x, otherwise images might get too blurry and this was something @jancborchardt wanted to avoid.
In fact it is only a shortcut for downloading, but case sensitive.

I have no hard feelings above this at all…
Let us have at least two additional opinions, best would be one from @nextcloud/designers, and then just do what the majority decides :-)

**Comment by jancborchardt** [comment]
Yes, as @tobiasKaminsky said it’s a kind of a shortcut for downloading. I think it’s quite useful, only coming in when you need it.

@ardevd if you never noticed that or never needed it, that’s good too, because it’s a progressive enhancement shortcut.
